app/data/repositories/CardsRepository.py

```diff
@@ -20,8 +22,11 @@ def create(deck_id, word, translation) -> Optional[Card]:
     card_id = c.lastrowid
 
     if card_id:
```

Comment:
You could go for having main context in repositories too, like here:
logger.info(
            f"Handling '/deck/{deck_id}/create_card' route, user is not authenticated")

context here is 'Saving card to database'

but as you wish, those existing logs are fine too

Reply:
They are fine :)